src/main/java/core/basesyntax/converter/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.converter;
+
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> converterToTransaction(List<String> inputData) {
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String data: inputData) {
+            if (data.equals("type,fruit,quantity")) {
+                continue;
+            }
+            String[] parts = data.split(",");
+            String operationCode = parts[0].trim();
+            String fruitName = parts[1].trim();
+            int quantity = Integer.parseInt(parts[2].trim());
+
+            // Determine the type of activity
+            FruitTransaction.Operation operation
+                    = FruitTransaction.Operation.fromCode(operationCode);
+
+            // We create a new object and add it to the list
+            FruitTransaction transaction = new FruitTransaction();
+            transaction.setOperation(operation);
+            transaction.setFruit(fruitName);
+            transaction.setQuantity(quantity);

```

Comment:
Consider using a constructor to initialize the FruitTransaction object instead of using setters. It's a good practice to make your objects immutable whenever possible. Immutable objects are simpler to understand and use, more secure and more efficient. You can create a constructor in the FruitTransaction class that takes operation, fruit and quantity as parameters and initializes the corresponding fields.